guides/image-sources/provisioning-image-finish.plantuml

```diff
@@ -0,0 +1,47 @@
+@startuml
+
+!include foreman.pstyle
+
+title Image-based provisioning with Finish script configuration
+
+actor       User
+participant "Provisioned\ninstance" as Host
+participant "Foreman" as Foreman
+participant "Foreman\nProxy" as Proxy
+participant "Infrastructure\ncloud" as Cloud
+participant DNS
+!if ($puppet)
+participant "Puppet\nserver" as Puppet
+!endif
+
+note over Foreman : Has an image\nwith SSH credentials\ndefined in compute resource
+note over Host : Powered off
+
+== Create host in Foreman ==
+
+User -> Foreman : Create host\nwith a cloud resource and the image
+group Template [Finish]
+   Foreman -> Proxy : Render the Finish script
+end
+Foreman -> Cloud : Create a new instance
+Cloud -> Foreman : Report the IP address
+Foreman -> Proxy : Create DNS records
+Proxy -> DNS : Forward DNS records
+Foreman -> Cloud : Start the instance
+Cloud -> Host : Boot
```

Comment:
```suggestion
Cloud -> Host : Boot
group Template [Finish]
   Foreman -> Foreman : Render the Finish script
end
```
Like this?

Reply:
What section? I don't want to remove the `group` because I use it as a semantic element to signify what kind of template is rendered.

Reply:
@stejskalleos Can you please explain why?

I think that the template has to be rendered somewhere and somehow for the workflow to function.